src/string_search.h

```diff
@@ -0,0 +1,597 @@
+// Copyright 2011 the V8 project authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef SRC_STRING_SEARCH_H_
+#define SRC_STRING_SEARCH_H_
+
+#include "node.h"
+#include <string.h>
+
+namespace node {
+namespace stringsearch {
+
+// Returns the maximum of the two parameters.
+template <typename T> T Max(T a, T b) { return a < b ? b : a; }
+
+static const uint32_t kMaxOneByteCharCodeU = 0xff;
+
+static inline int NonOneByteStart(const uint16_t *chars, int length) {
+  const uint16_t *limit = chars + length;
+  const uint16_t *start = chars;
```

Comment:
Correct, it should conform to the code style.  While you're here, can you make sure it uses `size_t` instead of `int` for string lengths, array offsets, etc.?  Google's insistence on using signed types for that is something I'd not like us to copy.